utils/common/src/main/java/org/apache/brooklyn/util/JavaGroovyEquivalents.java

```diff
@@ -66,8 +66,9 @@ public static String elvis(String preferred, String fallback) {
     public static String elvisString(Object preferred, Object fallback) {
         return elvis(asString(preferred), asString(fallback));
     }
+    @SuppressWarnings("unchecked")
     public static <T> T elvis(T preferred, T fallback) {
-        return groovyTruth(preferred) ? preferred : fallback;
+        return (T) fix(groovyTruth(preferred) ? preferred : fallback);
```

Comment:
@aledsage can we have a test with elvis(new GString(), new GString())?

The second, you are right,  I didn't check that it is `String` and do not need `fix`